File: StkExtensionPlugins/PostProcessing/Perl/pps_simple.pl
#
# this is the temp file used by STK
#
$inputfile = shift;

#
# open/read/close the file
#
open (INPUT, "$inputfile");
@inData = <INPUT>;
close INPUT;

#
# open the new file (overwite input file) for writing
#
open (OUTPUT, ">$inputfile");
$count = 1;
foreach (@inData){
	# do something with the data
	print OUTPUT $count++ . " " . $_;
	}

#
# close the output file
#
close OUTPUT;

#
# when the script ends, STK will disply whatever is in the $inputfile
#
exit;

File: StkExtensionPlugins/PostProcessing/Python/pps_simple.py
#
# this is the temp file used by STK
#
import sys

inputfile = sys.argv[1]

#
# open/read the file
#
content = []
with open(inputfile) as f:
    content = f.readlines()

#
# open the new file (overwite input file) for writing
#
outfile = open(inputfile, "w")
count = 1
for s in content:
    outfile.write(str(count) + " " + s),
    count += 1

#
# close the output file
#
outfile.close()
